astropy/modeling/core.py

```diff
@@ -801,6 +807,94 @@ def inverse(self):
         raise NotImplementedError("An analytical inverse transform has not "
                                   "been implemented for this model.")
 
+    @property
+    def bounding_box(self):
+        """
+        A `tuple` of length `n_inputs` defining the bounding box limits, or 
+        `None` for no bounding box. 
+
+        The default is `None`, unless ``bounding_box_default`` is defined.
+        `bounding_box` can be set manually to an array-like  object of shape 
+        ``(model.n_inputs, 2)``. For further usage, including how to set the
+        ``bounding_box_default``, see :ref:`bounding-boxes`
+
+        The limits are ordered according to the `numpy` indexing
+        convention, and are the reverse of the model input order,
+        e.g. for inputs ``('x', 'y', 'z')`` the ``bounding_box`` is defined:
+        
+        * for 1D: ``(x_low, x_high)``
+        * for 2D: ``((y_low, y_high), (x_low, x_high))``
+        * for 3D: ``((z_low, z_high), (y_low, y_high), (x_low, x_high))``
+
+        Examples
+        --------
+        Setting the bounding boxes for a 1D, 2D, and custom 3D model.
+
+        >>> from astropy.modeling.models import Gaussian1D, Gaussian2D, custom_model
+        >>> model_1d = Gaussian1D()
+        >>> model_2d = Gaussian2D(x_stddev=1, y_stddev=1)
+
+        Set the bounding box like:
+
+        >>> model_1d.bounding_box = (-5, 5)
+        >>> model_2d.bounding_box = ((-6, 6), (-5, 5))
+
+        For a user-defined 3D model:
+
+        >>> def const3d(x, y, z, amp=1):
+        ...    return amp
+        ...
+        >>> Const3D = custom_model(const3d)
+        >>> model_3d = Const3D()
+        >>> model_3d.bounding_box = ((-6, 6), (-5, 5), (-4, 4))
+
+        To reset the default:
+
+        >>> model_1d.bounding_box = 'auto'
+
+        To turn off the bounding box:
+
+        >>> model_1d.bounding_box = None
+
+        """
+
+        if self._bounding_box == 'auto':
+            return self.bounding_box_default()
+
+        else:
+            return self._bounding_box
+
+    @bounding_box.setter
+    def bounding_box(self, limits):
+        """
+        Assigns the bounding box limits. 
+        """
+
+        if limits == 'auto':
+            if not hasattr(self, 'bounding_box_default'):
+                warnings.warn('The default for this model is None.')
```

Comment:
Although actually I think it would be better if all models have `bounding_box_default` *defined*.  The default implementation should raise `NotImplementedError`, and this could would check for that.  Rather than storing the actual value 'auto' in `self._bounding_box` it should just set `self._bounding_box` to the result of `self.bounding_box_default()`)